File: server/thryve/src/main/kotlin/com/hatchgrid/thryve/newsletter/tag/application/TagResponse.kt
package com.hatchgrid.thryve.newsletter.tag.application

import com.hatchgrid.thryve.newsletter.tag.domain.Tag
import java.io.Serializable

data class TagResponse(
    val id: String,
    val name: String,
    val color: String,
    val workspaceId: String,
    val subscribers: List<String>? = emptyList(),
    val createdAt: String,
    val updatedAt: String?,
) : Serializable {
    companion object {
        private const val serialVersionUID: Long = 1L

        fun from(tag: Tag): TagResponse {
            return TagResponse(
                id = tag.id.value.toString(),
                name = tag.name,
                color = tag.color.value,
                workspaceId = tag.workspaceId.value.toString(),
                subscribers = tag.subscribers?.map { it.value },
                createdAt = tag.createdAt.toString(),
                updatedAt = tag.updatedAt?.toString(),
            )
        }
    }
}

File: client/apps/web/src/tag/domain/models/TagResponse.ts
export interface TagResponse {
	id: string; // UUID v4
	name: string;
	color: string;
	subscribers: ReadonlyArray<string> | string;
	createdAt?: string;
	updatedAt?: string;
}

export interface CreateTagRequest {
	name: string;
	color: string;
}

export interface UpdateTagRequest {
	name?: string;
	color?: string;
}
